components/shortcut/actions/create-story/create-story.js

```diff
@@ -0,0 +1,357 @@
+      optional: true,
+    },
+    storyType: {
+      type: "string",
+      label: "Story Type",
+      description: "The type of story (feature, bug, chore).",
+      options: [
+        "bug",
+        "chore",
+        "feature",
+      ],
+      default: "feature",
+      optional: true,
+    },
+    task: {
+      type: "object",
+      label: "Task",
+      description:
+        "A task object attached to the story. Each task object must have the following structure: `complete` which is a boolean, indicating whether the task is completed (defaults to `false`), `created_at` which defaults to the time/date the task is created but can be set to reflect another creation time/date, `description` as a description for the task, `external_id` a field can be set to another unique ID. In the case that the task has been imported from another tool, the ID in the other tool can be indicated here, `owner_ids` as an array of UUIDs for any members you want to add as owners on this new task, `updated_at` which defaults to the time/date the task was last updated in Shortcut but can be set to reflect another time/date.  See [CreateTaskParams](https://shortcut.com/api/rest/v3#CreateTaskParams) for more info.",
+      optional: true,
+    },
+    updatedAt: {
+      type: "string",
+      label: "Updated at Date",
+      description: "The time/date the story was updated.",
+      optional: true,
+    },
+    workflowStateId: {
+      type: "integer",
+      label: "Workflow State Id",
+      description: "The ID of the workflow state the story will be in.",
+      async options() {
+        const workflows = await this.shortcut.callWithRetry("listWorkflows");
+        return workflows.reduce(function (options, workflow) {
+          const hasState = get(workflow, [
+            "states",
+            "length",
+          ]);
+          if (hasState) {
+            workflow.states.map((state) => (options.push({
+              label: `${state.name} (${workflow.name})`,
+              value: `${state.id}`,
+            })));
+          }
+          return options;
+        }, []);
+      },
+      optional: true,
+    },
+  },
+  async run() {
+    const constraints = {
+      name: {
+        length: {
+          maximum: 512,
+        },
+      },
+      description: {
+        length: {
+          maximum: 100000,
+        },
+      },
+    };
+    if (this.externalLinks) {
+      constraints.externalLinks = {
+        type: "array",
+      };
+    }
+    if (this.fileIds) {
+      constraints.fileIds = {
+        type: "array",
+      };
+    }
+    if (this.followerIds) {
+      constraints.followerIds = {
+        type: "array",
+      };
+    }
+    if (this.linkedFileIds) {
+      constraints.linkedFileIds = {
+        type: "array",
+      };
+    }
+    if (this.ownerIds) {
+      constraints.ownerIds = {
+        type: "array",
+      };
+    }
```

Comment:
I think you can write the constraint in just one go with your own validator. That way you can remove all the `if` statements for `constraints`
```suggestion  
    const constraints = {
      name: {
        length: {
          maximum: 512,
        },
      },
      description: {
        length: {
          maximum: 100000,
        },
      },
      externalLinks: validateArrayWhenPresent,
      fileIds: validateArrayWhenPresent,
      followerIds: validateArrayWhenPresent,
      inkedFileIds: validateArrayWhenPresent,
      ownerIds: validateArrayWhenPresent,
    };
```
Then you can write your validator like this
```javascript
 function validateArrayWhenPresent(value) {
      if (value === null || value === undefined) {
        return;
      }
      return {
        type: "array",
      };
    }
```